### ISSUE TITLE
What to do with the apps directory?

### ISSUE THREAD

**Issue by amlalejini** [devosoft/Empirical#382]
The apps directory contains miscellaneous applications built with Empirical. These generally provide wonderful examples of using the library (and many are useful applications on their own). However, apps are not well-integrated into the library's testing infrastructure, and some of these applications are out of sync with the library (e.g., no longer compile). 

What do we want to do with this directory? Should the apps directory actually be in the examples directory? Should each app be moved to its own repository? Should we remove the apps that no longer compile?

**Comment by amlalejini** [comment]
If our goal is to ship Empirical as a general purpose scientific software library, we should better curate any applications/examples that come with the library (i.e., in the library's repository). Any applications we keep around should probably get worked into the testing system so that things can't silently get out of sync and then be a potential source of confusion for users looking for examples of how to use the library. 

What do folks think about the apps directory?

**Comment by amlalejini** [comment]
E.g., the 'Avida' app does not currently compile. Also, from the perspective of a new user, it has a potentially confusing name. Like, is it a re-implementation of the Avida ALIFE platform? No...

**Comment by amlalejini** [comment]
Also, what do each of these apps do, and how do you use them? (any that we keep should probably have an explanatory readme)

**Comment by mmore500** [comment]
Was about to hop on here to suggest we err towards shunting things into the OLD directory (and can always restore them if we want them) but then saw that you were already on it :joy_cat:

**Comment by mmore500** [comment]
adding the apps to the examples build or adding a new travis job sounds awesome

**Comment by mmore500** [comment]
my understanding is that the examples is organized more one-on-one with header files to provide micro usage demos and that the apps directory would also be more things that are informative/demonstrative (and not that we care about existing on their own) that show lots of different parts working together. so we should probably make a pass through the apps directory to throw stuff out/relocate to own repos that isn't informative or that we care about existing in its own right outside of a demo and keep the rest, we could even rename the folder "demos" which might be clearer for what we're getting at for it to be than "apps"

**Comment by amlalejini** [comment]
Yeah, good point about the distinction between examples and apps, and I love the idea of renaming the apps directory to demos. 

As per the #Empirical slack channel, it sounds like Charles is making a pass through the apps directory to curate a bit.
